lib/core.js

```diff
@@ -300,6 +300,15 @@ interface Iterable<T> {
     @@iterator(): Iterator<T>;
 }
 
+/* Generators */
+interface Generator<Y,R,N> {
+    @@iterator(): Iterator<Y>;
+    next(value?: N): IteratorResult<R|Y>;
+}
+
+// we use this signature when typing yield* expressions
+declare var $yieldDelegate: (<R,N>(g: Generator<*,R,N>, n:N) => R) & ((i: Iterable<*>, n:void) => void);
```

Comment:
The [test for delegate yields to iterables](https://github.com/facebook/flow/pull/534/files#diff-235c03962cd0739e93a3d5d899a2a634R96) fails, and I think this intersection type is to blame. The unexpected error states that Array<number> doesn't have a `next` method, which is a part of the Generator interface. My expectation was that flow would fail through and use the second case when that happens, instead of spitting out an error.